INSTALLING.md

```diff
@@ -0,0 +1,306 @@
+> WARNING: Using the --with-extra-version option during the configuration phase can break the sys.get_host_os() output; we do not recommend including it.
+
+Build and install the extensions because `uuid-ossp.so` is a runtime dependency for Babelfish:
+
+```sh
+cd ${PG_SRC}/contrib && make -j ${JOBS} && sudo make install
+```
+
+### Compile the ANTLR parser generator 
+
+Use the following commands to compile the ANTLR parser generator and copy the runtime to the PostgreSQL library location:
+
+``` sh
+export cmake=$(which cmake)
+          
+sudo cp /usr/local/lib/libantlr4-runtime.so.${ANTLR4_VERSION} ${BABELFISH_HOME}/lib
+           
+cd ${PG_SRC}/contrib/babelfishpg_tsql/antlr 
+cmake -Wno-dev .
+make all
+``` 
+
+### Compile the contrib modules and build Babelfish
+
+Now, it is time to compile the contrib modules and build Babelfish. Use the command:
+
+```sh
+cd ${PG_SRC}/contrib
+for ext in babelfishpg_common babelfishpg_money babelfishpg_tds babelfishpg_tsql 
+do
+    cd $ext
+    make -j ${JOBS}
+    sudo make PG_CONFIG=${PG_CONFIG} install
+    cd ..
+done
+```
+
+## Setting up the PostgreSQL modified instance
+
+The steps required to create a new cluster and start the service are very similar to the steps required by a community PostgreSQL installation:
+
+```sh
+sudo mkdir -p /var/lib/babelfish/1.4.0
+
+sudo adduser postgres --home /var/lib/babelfish
+```
+
+Ensure that the related folders hold the permissions for the service user:
+
+```sh
+sudo chown -R postgres: /opt/babelfish/
+sudo chown -R postgres: /var/lib/babelfish/
+```
+
+Switch to the `postgres` user (a non-superuser) and start the cluster:
+
+```sh
+sudo su - postgres
+export BABELFISH_HOME=/opt/babelfish/1.4.0
+export BABELFISH_DATA=/var/lib/babelfish/1.4.0/data
+```
+
+### Initiating the Data directory
+
+#### Initiate data with trustable connection for development
+
+If you would like to create a local cluster for testing purposes, you can configure trust authentication when initializing the database to simplify authentication. The `--auth-host=trust` flag will create the cluster using trust authentication, and should not be included if you are creating an instance that will contain sensitive information:
+
+```sh
+${BABELFISH_HOME}/bin/initdb -D ${BABELFISH_DATA}/ -E "UTF8" --auth=trust --auth-host=trust --auth-local=trust
+```
+
+#### Initiate data with specific HBA configuration
+
+For production or isolated environments, it is recommended to specify the allowed IP addresses that can access the system and a secure authentication method such as `md5` in the `pg_hba.conf`. For other supported methods see [Authentication Methods documentation](https://www.postgresql.org/docs/14/auth-methods.html).
+
+```sh
+${BABELFISH_HOME}/bin/initdb -D ${BABELFISH_DATA}/ -E "UTF8"
+```
+
+Once the data directory is initialized, edit the `${BABELFISH_DATA}/pg_hba.conf` file following the [vanilla Postgres HBA configuration](https://www.postgresql.org/docs/14/auth-pg-hba-conf.html).
```

Comment:
Shouldn't this be https://www.postgresql.org/docs/13/auth-pg-hba-conf.html

Reply:
Updated!